notary-server/Dockerfile

```diff
@@ -1,25 +1,35 @@
-FROM golang:1.10.3-alpine
+FROM alpine:latest
+
 ENV TAG v0.6.1
 ENV NOTARYPKG github.com/theupdateframework/notary
-WORKDIR /go/src/${NOTARYPKG}
-RUN apk --update add tar
-RUN wget -O notary.tar.gz "https://api.github.com/repos/theupdateframework/notary/tarball/${TAG}" && tar xzf notary.tar.gz --strip-components=1
-RUN go install \
-    -ldflags "-w -X ${NOTARYPKG}/version.GitCommit=`wget -qO- https://api.github.com/repos/theupdateframework/notary/tags | grep -A 5 ${TAG} | grep sha | awk '{print substr($2,2,8)}'` -X ${NOTARYPKG}/version.NotaryVersion=`cat NOTARY_VERSION`" \
-    ${NOTARYPKG}/cmd/notary-server
+ENV NOTARYREPO https://api.github.com/repos/theupdateframework/notary
+ENV INSTALLDIR /notary/server
+EXPOSE 4443
 
+WORKDIR /usr/lib/go/src/${NOTARYPKG}
 
-FROM alpine:latest
-EXPOSE 4443
-WORKDIR /notary/server
+RUN apk --no-cache add tar go musl-dev && \
+    wget -O notary.tar.gz "${NOTARYREPO}/tarball/${TAG}" && \
+    tar xzf notary.tar.gz --strip-components=1 && \
+    go install \
+	-ldflags "-w -X ${NOTARYPKG}/version.GitCommit=`wget -qO- ${NOTARYREPO}/tags | \
+		grep -A 5 ${TAG} | grep sha | \
+		awk '{print substr($2,2,8)}'` \
+		-X ${NOTARYPKG}/version.NotaryVersion=`cat NOTARY_VERSION`" \
+	${NOTARYPKG}/cmd/notary-server && \
```

Comment:
This scraping via `wget` seems really hacky -- why are we going to all this trouble?

If it's important for `version.GitCommit` to be set to a value that isn't simple like `$TAG` (which is technically a valid Git ref, and is the ref we're building from), then why not instead install `git`, do a `git clone --depth 1`, and then use the existing `Makefile` to let the upstream build set the values appropriately itself?

Here's an example `Dockerfile` I've successfully tested (on both amd64 and arm64) that runs and generates a ~21MB image (~4MB for `alpine:3.8` and ~17MB for `notary-server`), and provides the expected values in `notary-server -debug`:

```dockerfile
FROM alpine:3.8

RUN adduser -D -H -g "" notary

ENV PATH=$PATH:/notary/server
WORKDIR /notary/server

ENV NOTARYPKG github.com/theupdateframework/notary
ENV TAG v0.6.1

RUN set -eux; \
	\
	apk add --no-cache --virtual .build-deps git go make musl-dev; \
	\
	export GOPATH=/go GOCACHE=/go/cache; \
	mkdir -p "$GOPATH/src/$NOTARYPKG"; \
	git clone -b "$TAG" --depth 1 "https://$NOTARYPKG" "$GOPATH/src/$NOTARYPKG"; \
	\
	make -C "$GOPATH/src/$NOTARYPKG" PREFIX=. ./bin/static/notary-server; \
	cp -vL "$GOPATH/src/$NOTARYPKG/bin/static/notary-server" ./; \
	rm -rf "$GOPATH"; \
	\
	apk del --no-network .build-deps

EXPOSE 4443

COPY ./server-config.json .
COPY ./entrypoint.sh .

USER notary

ENTRYPOINT [ "entrypoint.sh" ]
CMD [ "notary-server", "--help" ]
```

I also don't understand why this and https://github.com/docker/notary-official-images/pull/11 are separate PRs (since it's effectively the same exact change), but the same approach will likely work in exactly the same way over on `notary-signer`.

Reply:
> ... the same approach will likely work in exactly the same way over on `notary-signer`.

Confirmed -- copied, replaced all `server` with `signer` and everything appears to work.

Reply:
One step at a time.  This PR moves everything in the right direction (removes multi-stage builds) - it does not make things perfect.  In the projects I normally work in, commits are expected to contain only one functional change.

They are separate PRs because I wanted to see if the change would be well received before spending even more effort on something that would have been NACKed.  Ultimately I'm happy to merge them.